userland/libtock/FXOS8700CQ.c

```diff
@@ -0,0 +1,49 @@
+#include <stdio.h>
+#include "FXOS8700CQ.h"
+#include "math.h"
+
+struct fx0_data {
+  int x;
+  int y;
+  int z;
+  bool fired;
+};
+
+// internal callback for faking synchronous reads
+static void FXOS8700CQ_cb(int x, int y, int z, void* ud) {
+  struct fx0_data* result = (struct fx0_data*) ud;
+  printf("callback x= %i y = %i z = %i\n", x, y, z); 
+  result->x = x;
+  result->y = y; 
+  result->z = z; 
+  result->fired = true;
+}
+
+double FXOS8700CQ_read_accel_mag() {
+  struct fx0_data result = { .fired = false };
+  int err;
+
+  err = FXOS8700CQ_subscribe(FXOS8700CQ_cb, (void*)(&result));
+  if (err < 0) {
+    return err;
+  }
+
+  err = FXOS8700CQ_start_accel_reading();
+  if (err < 0) {
+    return err;
+  }
+
+  yield_for(&result.fired);
+
+  // TODO add sqrt 
```

Comment:
Just add a short explanation for what this is blocking on. E.g., "add sqrt once software floating-point is supported in userspace. Currently would crash."